dist/web3-light.js

```diff
@@ -3495,6 +3516,201 @@ module.exports = ICAP;
+
+        // DE-CHUNKER
+        var dechunkedData = data
+            .replace(/\}\{/g,'}|--|{') // }{
+            .replace(/\}\]\[\{/g,'}]|--|[{') // }][{
+            .replace(/\}\[\{/g,'}|--|[{') // }[{
+            .replace(/\}\]\{/g,'}]|--|{') // }]{
+            .split('|--|');
+
+
+        dechunkedData.forEach(function(data){
+
+            // prepend the last chunk
+            if(_this.lastChunk)
+                data = _this.lastChunk + data;
+
+            var result = data,
+                id = null;
+
+            try {
+                result = JSON.parse(result);
+
+            } catch(e) {
+
+                _this.lastChunk = data;
+
+                // start timeout to cancel all requests
+                clearTimeout(_this.lastChunkTimeout);
+                _this.lastChunkTimeout = setTimeout(function(){
+                    _this.timeout();
+                    throw errors.InvalidResponse(result);        
+                }, 1000 * 15);
+
+                return;
+            }
+
+            // cancel timeout and set chunk to null
+            clearTimeout(_this.lastChunkTimeout);
+            _this.lastChunk = null;
+
+            // get the id which matches the returned id
+            if(utils.isArray(result)) {
+                result.forEach(function(load){
+                    if(_this.responseCallbacks[load.id])
+                        id = load.id;
+                });
+            } else {
+                id = result.id;
+            }
+
+            // fire the callback
+            if(_this.responseCallbacks[id]) {
+                _this.responseCallbacks[id](null, result);
+                delete _this.responseCallbacks[id];
+            }
+        });
+    });
+};
+
+/**
+Get the adds a callback to the responseCallbacks object,
+which will be called if a response matching the response Id will arrive.
+
+@method _getResponse
+*/
+IpcProvider.prototype._getResponse = function(payload, callback) {
+    var id = payload.id || payload[0].id;
+    var method = payload.method || payload[0].method;
+
+    this.responseCallbacks[id] = callback;
+    this.responseCallbacks[id].method = method;
+};
+
+/**
+Timeout all requests when the end/error event is fired
+
+@method _timeout
+*/
+IpcProvider.prototype._timeout = function() {
+    for(var key in this.responseCallbacks) {
+        if(this.responseCallbacks.hasOwnProperty(key)){
+            this.responseCallbacks[key](errorTimeout.replace('__id__', key).replace('__method__', this.responseCallbacks[key].method));
+            delete this.responseCallbacks[key];
+        }
+    }
+};
+
+
+/**
+Check if the current connection is still valid.
+
+@method isConnected
+*/
+IpcProvider.prototype.isConnected = function() {
+    var _this = this;
+
+    // try reconnect, when connection is gone
+    setTimeout(function(){
+        if(!_this.connection.writable)
+            _this.connection.connect({path: _this.path});
+    }, 0);
+
+    return !!this.connection.writable;
```

Comment:
I commented on this before. Method called `isConnected` should not handle reconnection.